lib/audited/sweeper.rb

```diff
@@ -18,7 +18,7 @@ def around(controller)
     end
 
     def current_user
-      controller.send(Audited.current_user_method) if controller.respond_to?(Audited.current_user_method, true)
+      lambda { controller.send(Audited.current_user_method) if controller.respond_to?(Audited.current_user_method, true) }
```

Comment:
Yeah, I was initially planning to store the controller but I didn't think it was a good idea to have the Audit model calling methods directly on the controller. I preferred to keep the controller-specific calls and logic in this class.

> ... then move the remaining code to the sweeper

Which code's that? Do you mean to the model?